src/node/createSetupServer.ts

```diff
@@ -137,6 +137,10 @@ export function createSetupServer(
         )
       },
 
+      listHandlers() {
+        return currentHandlers
```

Comment:
We need to be cautious exposing internals, and `currentHandlers` is internals. I fear that developers will be tempted to meddle with this array directly. That should never happen. They should use explicit APIs like `use` or `resetHandlers` to affect the list of handlers, and `listHandlers()` must be purely readonly. 

Maybe let's make this array readonly before exposing it?

```ts
// First, guard the array on the type level.
listHandlers(): ReadonlyArray<RequestHandler> {
  // Second, prevent mutations on runtime.
  const immutableHandlers = [...currentHandlers]
  Object.freeze(immutableHandlers)
  return immutableHandlers
}
```